File: HackerRank Solutions/Interview Preparation Kit/String Manipulation/Strings Making Anagrams.py
#!/bin/python3

import math
import os
import random
import re
import sys

CHARS = 26
# Complete the makeAnagram function below.
def makeAnagram(a, b):
    count = [0]*CHARS
    i=0
    while(i < len(a)):
        count[ord(a[i]) - ord('a')] += 1
        i += 1
    i=0
    while(i < len(b)):
        count[ord(b[i]) - ord('a')] -= 1
        i += 1
    result=0
    for i in range(26):
        result+= abs(count[i])
    return result;

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    a = input()

    b = input()

    result = makeAnagram(a, b)

    fptr.write(str(result) + '\n')

    fptr.close()

File: HackerRank Solutions/Interview Preparation Kit/String Manipulation/Strings Making Anagrams.cpp
#include <bits/stdc++.h>
using namespace std;

const int CHARS = 26;
// Complete the makeAnagram function below.
int makeAnagram(string a, string b) {
    int count[CHARS] = {0};
    for(int i=0; a[i]!='\0'; i++){
        count[a[i]-'a']++;
    }
    for(int i=0; b[i]!='\0'; i++){
        count[b[i]-'a']--;
    }
    int result = 0;
    for(int i=0; i<26; i++){
        result+=abs(count[i]);
    }
    return result;
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));
    string a;
    getline(cin, a);
    string b;
    getline(cin, b);
    int res = makeAnagram(a, b);
    fout << res << "\n";
    fout.close();
    return 0;
}
